src/transaction.js

```diff
@@ -35,6 +44,9 @@ function Transaction(id, callback) {
  * @return {Promise}
  */
 Transaction.prototype.execute = function() {
+  if (this.promise.isRejected()) {
+    return this.promise;
+  }
   return this._callback.apply(this, arguments);
```

Comment:
Small, small question?
Could we...
```js
this._callback.apply(this, arguments);
return this.promise;
```

Or wait actually you don't wanna run callback twice, right?
So maybe we should rather:
```js
if (!this.promise.isPending()) {
  return this.promise;
}
return this._callback.apply(this, arguments);
```

?